WinCompatibilityPack/WinCompatibilityPack.psm1

```diff
@@ -7,6 +7,32 @@ using namespace System.Management.Automation.Runspaces
 
 Set-StrictMode -Version latest
 
+#region .Net Windows Compatibility Pack assemblies
+
+$runtimePath = ""
+if ($PSVersionTable.PSEdition -eq 'Desktop')
+{
+    $runtimePath = "net472"
+}
+elseif ($PSVersionTable.PSEdition -eq 'Core')
+{
+    $runtimePath = "netcoreapp20"
+}
+else
+{
+    throw "Only PSEdition 'Desktop' and 'Core' is supported"
+}
+
+$currentPath = Split-Path $MyInvocation.MyCommand.Path -Parent
+$runtimePath = Join-Path -Path $currentPath -ChildPath $runtimePath
+
+foreach ($assembly in (Get-ChildItem -Path $runtimePath))
+{
+    Add-Type -Path $assembly.FullName
```

Comment:
I remember now what the issue is with doing it this way. On Core, when you do

```powershell
Import-Module .\bin\WinCompatibilityPack.psd1 -Force
```

the `Add-Type` blows up with a bunch of errors like

```none
Add-Type : Could not load file or assembly 'Microsoft.Win32.Registry.AccessControl, Version=4.0.3.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'.
At C:\Git\WindowsPowerShellCompatibilityPack\bin\WinCompatibilityPack.psm1:31 char:5
+     Add-Type -Path $assembly.FullName
+     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ CategoryInfo          : NotSpecified: (:) [Add-Type], FileLoadException
+ FullyQualifiedErrorId : System.IO.FileLoadException,Microsoft.PowerShell.Commands.AddTypeCommand
```

This doesn't happen on a normal import, just with force. This doesn't seem to be an issue with 5.1 as i can import with `-Force` without issues multiple times. I suspect the failures are for assemblies that are already as we ship `Microsoft.Win32.Registry.AccessControl.dll` with Core.

It would probably be better to use the `$PSEdition` gate in the `.psd1` for RequiredAssemblies as it properly handles this. https://docs.microsoft.com/en-us/powershell/gallery/concepts/module-psedition-support#option-2-use-psedition-variable-in-the-psd1-file-to-load-the-proper-dlls-and-nestedrequired-modules